src/main/java/edu/wpi/grip/ui/OperationView.java

```diff
@@ -88,5 +88,28 @@ public void initialize(URL url, ResourceBundle resourceBundle) {
             Step step = new Step(this.eventBus, this.operation);
             this.eventBus.post(new StepAddedEvent(step));
         });
+
+
+        // Ensures that when this element is hidden that it also removes its size calculations
+        this.managedProperty().bind(this.visibleProperty());
+    }
+
+
+    /**
+     * Uses the search string from the palette and allows this element to determine if it should be visible
+     *
+     * @param searchString The search string that should be used to determine visibility
+     */
+    public void determineVisibility(final String searchString) {
```

Comment:
Also, keep in mind that  `String::contains` will probably not be our final search function.  Assuming we eventually have a more advanced fuzzy search:

1. The proposed design doesn't allow us to sort by relevance to a search (so if I type "eige", eigenvalue related operations show up before "addWeighted")
2. The proposed design allow us to search with more advanced options (search by name, by description, by category, using regex) without adding more methods to let `OperationView` access the values in the views inside of the `PaletteView`.